numpy/core/tests/test_multiarray.py

```diff
@@ -5475,9 +5476,29 @@ def test_basic(self, byteorder, dtype):
         buf = x.tobytes()
         assert_array_equal(np.frombuffer(buf, dtype=dt), x.flat)
 
+    def test_array_base(self):
+        arr = np.arange(10)
+        new = np.frombuffer(arr)
+        # We currently special case arrays to ensure they are used as a base.
+        # This could probably be changed (removing the test).
+        assert new.base is arr
+
     def test_empty(self):
         assert_array_equal(np.frombuffer(b''), np.array([]))
 
+    def test_mmap_close(self):
```

Comment:
@mattip the test just needs to be skipped on PyPy, or is there something else going on?

Reply:
PyPy does not track the number of exported mmap references, and so the same segfault that occurs in issue gh-9537 will still happen in PyPy. I opened an issue on PyPy but it is complicated to properly support this. https://foss.heptapod.net/pypy/pypy/-/issues/3724

Reply:
Thanks, makes sense since it adds another level of reference counting effectively.